Editor/Window/ManagedEC2Deployment.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Threading.Tasks;
+using AmazonGameLift.Editor;
+using UnityEngine;
+
+namespace AmazonGameLift.Editor
+{
+    public class ManagedEC2Deployment
+    {
+        private readonly DeploymentSettings _model;
+        private readonly FleetParameters _parameters;
+        
+        internal ManagedEC2Deployment(DeploymentSettings model, FleetParameters parameters)
+        {
+            _model = model;
+            _parameters = parameters;
+        }
+        
+        public void UpdateModelFromParameters()
+        {
+            _model.FleetName = _parameters.FleetName;
+            _model.BuildName = _parameters.BuildName;
+            _model.LaunchParameters = _parameters.LaunchParameters;
+            _model.BuildFolderPath = _parameters.GameServerFolder;
+            _model.BuildFilePath = _parameters.GameServerFile;
+            _model.BuildOperatingSystem = _parameters.OperatingSystem;
+        }
+        
+        public void StartDeployment()
+        {
+            UpdateModelFromParameters();
+            _model.GameName = Application.productName.Substring(0, 12);
+            if (!_model.CanDeploy) return;
+            _model.Save();
+            _model.StartDeployment(ConfirmChanges).ContinueWith(task =>
+            {
+                if (task.IsFaulted)
+                {
+                    Debug.LogException(task.Exception);
+                }
+            });
+        }
+
+        public async Task DeleteDeployment()
+        {
+            var waiter = new DeploymentStatusPoller(CoreApi.SharedInstance);
+            waiter.InfoUpdated += () => { _model.RefreshCurrentStackInfo(); };
+            await _model.DeleteDeployment();
+            await waiter.WaitUntilDone(_model);
+        }
```

Comment:
Can we move DeleteDeployment to `DeploymentSettings` instead? imo it should be an async method, which callers can wait on until completed. This seems like an internal feature that `DeploymentSettings` should offer, rather than have to implement it outside.

Reply:
Updated